src/main/java/org/tron/core/services/interfaceOnSolidity/RpcApiServiceOnSolidity.java

```diff
@@ -0,0 +1,391 @@
+      } else {
+        responseObserver.onNext(null);
+      }
+      responseObserver.onCompleted();
+    }
+
+    @Override
+    public void getBlockByNum2(NumberMessage request,
+        StreamObserver<BlockExtention> responseObserver) {
+      long num = request.getNum();
+      if (num >= 0) {
+        Block reply = walletOnSolidity.getBlockByNum(num);
+        responseObserver.onNext(block2Extention(reply));
+      } else {
+        responseObserver.onNext(null);
+      }
+      responseObserver.onCompleted();
+    }
+
+    @Override
+    public void getTransactionCountByBlockNum(NumberMessage request,
+        StreamObserver<NumberMessage> responseObserver) {
+      NumberMessage.Builder builder = NumberMessage.newBuilder();
+      try {
+        Block block = dbManager.getBlockById(
+            dbManager.getBlockIndexStore().getOnSolidity(request.getNum())).getInstance();
+        builder.setNum(block.getTransactionsCount());
+      } catch (StoreException e) {
+        logger.error(e.getMessage());
+        builder.setNum(-1);
+      }
+      responseObserver.onNext(builder.build());
+      responseObserver.onCompleted();
+    }
+
+    @Override
+    public void getTransactionById(BytesMessage request,
+        StreamObserver<Transaction> responseObserver) {
+      ByteString id = request.getValue();
+      if (null != id) {
+        Transaction reply = walletOnSolidity.getTransactionById(id);
+
+        responseObserver.onNext(reply);
+      } else {
+        responseObserver.onNext(null);
+      }
+      responseObserver.onCompleted();
+    }
+
+    @Override
+    public void getTransactionInfoById(BytesMessage request,
+        StreamObserver<TransactionInfo> responseObserver) {
+      ByteString id = request.getValue();
+      if (null != id) {
+        TransactionInfo reply = walletOnSolidity.getTransactionInfoById(id);
+
+        responseObserver.onNext(reply);
+      } else {
+        responseObserver.onNext(null);
+      }
+      responseObserver.onCompleted();
+    }
+
+    @Override
+    public void generateAddress(EmptyMessage request,
+        StreamObserver<AddressPrKeyPairMessage> responseObserver) {
+      ECKey ecKey = new ECKey(Utils.getRandom());
+      byte[] priKey = ecKey.getPrivKeyBytes();
+      byte[] address = ecKey.getAddress();
+      String addressStr = Wallet.encode58Check(address);
+      String priKeyStr = Hex.encodeHexString(priKey);
+      AddressPrKeyPairMessage.Builder builder = AddressPrKeyPairMessage.newBuilder();
+      builder.setAddress(addressStr);
+      builder.setPrivateKey(priKeyStr);
+      responseObserver.onNext(builder.build());
+      responseObserver.onCompleted();
+    }
+  }
+
+  @Override
+  public void stop() {
+    if (apiServer != null) {
+      apiServer.shutdown();
+    }
+  }
+
+  /**
+   * ...
+   */
+  public void blockUntilShutdown() {
+    if (apiServer != null) {
+      try {
+        apiServer.awaitTermination();
+      } catch (InterruptedException e) {
```

Comment:
![Bug](https://sonarsource.github.io/sonar-github/bug.png 'Bug') Bug: Either re-interrupt this method or rethrow the "InterruptedException". ([squid:S2142](https://sonarcloud.io/organizations/tron-zhaohong/rules#rule_key=squid%3AS2142))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=java-tron&pullRequest=1691&issues=AWbtIg5FW3Y69bz58AlN&open=AWbtIg5FW3Y69bz58AlN)

[ISSUE_KEY]: AWbtIg5FW3Y69bz58AlN